src/sentry/workflow_engine/processors/delayed_workflow.py

```diff
@@ -112,23 +122,24 @@ def fetch_workflows_and_environments(
     return workflow_ids_to_workflows, workflows_to_envs
 
 
-def fetch_active_data_condition_groups(
+def fetch_detectors(detector_ids: list[int]) -> DetectorMapping:
+    detector_ids_to_detectors: DetectorMapping = {}
+    detectors = list(Detector.objects.filter(id__in=detector_ids))
+
+    for detector in detectors:
+        detector_ids_to_detectors[detector.id] = detector
+
+    return detector_ids_to_detectors
```

Comment:
is this so we can eventually get the DCG from the detector? (i wonder if we could just use the id to get the conditions themselves and skip loading the whole detector)

Reply:
hmm it's possible i don't actually need this, i suppose i don't need the Detector for WorkflowJob

Reply:
i'm actually only using `process_data_condition_group` instead of invoking the detector handler

Reply:
on third thought, i should have the `Detector` in order to invoke the next part of workflow engine if the `Detector` DCG passes (`process_workflows`)

Reply:
removed this function